utils/getClaimDetails.tsx

```diff
@@ -0,0 +1,129 @@
+}
+
+// Pairs of user-facing translation strings
+export const programExtensionPairs = {
+  UI: {
+    programType: 'claim-details:program-type.ui',
+    extensionType: '',
+  },
+  PEUC: {
+    programType: 'claim-details:program-type.ui',
+    extensionType: 'claim-details:extension-type.peuc-1',
+  },
+  PEUX: {
+    programType: 'claim-details:program-type.ui',
+    extensionType: 'claim-details:extension-type.peuc-2',
+  },
+  PEUY: {
+    programType: 'claim-details:program-type.ui',
+    extensionType: 'claim-details:extension-type.peuc-2-augmentation',
+  },
+  FEDED: {
+    programType: 'claim-details:program-type.ui',
+    extensionType: 'claim-details:extension-type.fed-ed',
+  },
+  TRA: {
+    programType: 'claim-details:program-type.ui',
+    extensionType: 'claim-details:extension-type.tra',
+  },
+  TRAAdditional: {
+    programType: 'claim-details:program-type.ui',
+    extensionType: 'claim-details:extension-type.tra-additional',
+  },
+  TE: {
+    programType: 'claim-details:program-type.ui',
+    extensionType: 'claim-details:extension-type.te',
+  },
+  DUA: {
+    programType: 'claim-details:program-type.dua',
+    extensionType: '',
+  },
+  PUA: {
+    programType: 'claim-details:program-type.pua',
+    extensionType: '',
+  },
+}
+
+/**
+ * Given a ProgramType string by the API gateway, return a pair of user-facing translation strings.
+ */
+export function getProgramExtensionPair(apiString: string): programExtensionPairType {
+  for (const [id, pair] of Object.entries(programExtensionPairs)) {
+    if (apiString === programTypeNames[id]) {
+      return pair
+    }
+  }
+  // If no known mapping is found, throw an error.
+  throw new Error('Unknown Program Type')
+}
+
+/**
+ * Format dates.
+ */
+export function formatDate(dateString: string): string {
+  const parsedDate = parse(dateString, "yyyy-MM-dd'T'HH:mm:ss", new Date())
+  return format(parsedDate, 'M/d/yyyy')
+}
+
+/**
+ * Constrtuct the benefit year string.
+ */
+export function buildBenefitYear(start: string, end: string): string {
+  return `${formatDate(start)} - ${formatDate(end)}`
+}
+
+/**
+ * Format currency.
+ */
+export function formatCurrency(amount: number): string {
+  return new Intl.NumberFormat('en-EN', { style: 'currency', currency: 'USD', minimumFractionDigits: 2 }).format(amount)
+}
+
+/**
+ * Get Claim Details content.
+ */
+export default function getClaimDetails(rawDetails: ClaimDetailsResult): ClaimDetailsContent {
+  // Get programType and extensionType.
+  const pair: programExtensionPairType = getProgramExtensionPair(rawDetails.programType)
+  const benefitYear = buildBenefitYear(rawDetails.benefitYearStartDate, rawDetails.benefitYearEndDate)
+
+  return {
+    programType: pair.programType,
+    benefitYear: benefitYear,
+    claimBalance: formatCurrency(rawDetails.claimBalance),
+    weeklyBenefitAmount: formatCurrency(rawDetails.weeklyBenefitAmount),
+    lastPaymentIssued: formatDate(rawDetails.lastPaymentIssued),
```

Comment:
Should we also be pulling out the `lastPaymentAmount`?

Reply:
See https://github.com/cagov/ui-claim-tracker/pull/307#discussion_r669174902